src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventLogger.java

```diff
@@ -0,0 +1,241 @@
+package com.google.cloud.bigquery.dwhassessment.hooks.logger;
+
+import static com.google.cloud.bigquery.dwhassessment.hooks.logger.LoggerVarsConfig.HIVE_QUERY_EVENTS_BASE_PATH;
+import static com.google.cloud.bigquery.dwhassessment.hooks.logger.LoggerVarsConfig.HIVE_QUERY_EVENTS_QUEUE_CAPACITY;
+import static com.google.cloud.bigquery.dwhassessment.hooks.logger.LoggerVarsConfig.HIVE_QUERY_EVENTS_ROLLOVER_CHECK_INTERVAL;
+import static com.google.cloud.bigquery.dwhassessment.hooks.logger.LoggingHookConstants.QUERY_EVENT_SCHEMA;
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Clock;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.RejectedExecutionException;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.commons.compress.utils.IOUtils;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.UUID;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.QueryPlan;
+import org.apache.hadoop.hive.ql.hooks.HookContext;
+import org.apache.hadoop.hive.ql.hooks.HookContext.HookType;
+import org.apache.hive.common.util.ShutdownHookManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Main handler of query events. Processes the captured events and writes resulting records to a
+ * file.
+ */
+public class EventLogger {
+  private static final Logger LOG = LoggerFactory.getLogger(EventLogger.class);
+
+  private static final int MAX_RETRIES = 2;
+  private static final int SHUTDOWN_WAIT_TIME = 5;
+  private static final int QUERY_EVENTS_QEUUE_DEFAULT_SIZE = 64;
+  private static final Duration DEFAULT_ROLLOVER_TIME_MILLISECONDS = Duration.of(600, SECONDS);
```

Comment:
1. I would suggest renaming to `DEFAULT_ROLLOVER_DURATION` as the value is not related to milliseconds.
2. The `Duration.of(600, SECONDS)` can be simplified to `Duration.of(10, MINUTES)`.